src/main/java/io/cdap/plugin/cdc/transform/CDCStreamingToRecordTransform.java

```diff
@@ -0,0 +1,213 @@
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) throws IllegalArgumentException {
+    super.configurePipeline(pipelineConfigurer);
+    Schema inputSchema = pipelineConfigurer.getStageConfigurer().getInputSchema();
+    config.validate(inputSchema);
+    try {
+      pipelineConfigurer.getStageConfigurer().setOutputSchema(Schema.parseJson(config.schema));
+    } catch (IOException e) {
+      throw new IllegalArgumentException("Output schema cannot be parsed.", e);
+    }
+  }
+
+  @Override
+  public void transform(StructuredRecord input, Emitter<StructuredRecord> emitter) throws Exception {
+
+    for (Schema.Field field : input.getSchema().getFields()) {
+      String name = field.getName();
+      //LOG.info("Local field name " + field.getName());
+      //LOG.info("Field type " + field.getSchema().toString());
+
+      if (input.get(name) != null && name.equals("dml")) {
+
+        //LOG.info("Preparing record");
+        //LOG.info(input.get(name).toString());
+        StructuredRecord sr = input.get(name);
+
+        try {
+          StructuredRecord out = getDMLRecord(sr);
+          emitter.emit(out);
+        } catch (Exception ex) {
+        }
+
+      }
+    }
+
+  }
+
+  private StructuredRecord getDMLRecord(StructuredRecord input) throws Exception {
+
+    Schema outputSchema = null;
+    Schema outSchema = null;
+
+    for (Schema.Field field : input.getSchema().getFields()) {
+      String name = field.getName();
+
+      if (name.equals("rows_schema")) {
+
+        outSchema = Schema.parseJson(input.get(name).toString());
+
+      }
+    }
+
+    List<Schema.Field> fields = new ArrayList<>(outSchema.getFields().size() + 1);
+    fields.addAll(outSchema.getFields());
+    fields.add(Schema.Field.of("CDC_OP_TYPE", Schema.nullableOf(Schema.of(Schema.Type.STRING))));
+    fields.add(Schema.Field.of("CHANGE_TRACKING_VERSION", Schema.nullableOf(Schema.of(Schema.Type.STRING))));
+
+    outputSchema = Schema.recordOf(outSchema.getRecordName() + ".added", fields);
+
+    //LOG.info("OutputSchema of new record " + outputSchema.toString());
+
+    StructuredRecord.Builder builder = StructuredRecord.builder(outputSchema);
+
+    try {
+
+      for (Schema.Field field : input.getSchema().getFields()) {
+
+        String name = field.getName();
+
+        if (name.equals("rows_values")) {
+          Gson gson = new Gson();
+          JsonReader reader = new JsonReader(new StringReader(input.get("rows_values").toString()));
+          reader.setLenient(true);
+          Map<String, Object> rowsValues = new HashMap<>();
+          rowsValues = gson.fromJson(reader, Map.class);
+
+
+          for (Map.Entry<String, Object> entry : rowsValues.entrySet()) {
+
+            String type = "string";
+
+            for (Schema.Field fieldtype : outSchema.getFields()) {
+              if (fieldtype.getName().equals(entry.getKey())) {
+
+                type = fieldtype.getSchema().toString().split("\"")[1];
```

Comment:
there are methods in Schema to get the type and logical type. There should not be manual schema json parsing like this being done.

Reply:
Its in not quite like that. We are using JTDS, and it have bad behaviour when parsing date time 2 MSSQL data type. It converts it incorrectly, so it was handled in this kind a way.I agree it should be changed